das/timeout_wrapper.go

```diff
@@ -12,27 +12,42 @@ import (
 	"github.com/offchainlabs/nitro/arbstate"
 )
 
-type TimeoutWrapper struct {
+type ReaderTimeoutWrapper struct {
+	t time.Duration
+	DataAvailabilityServiceReader
+}
+type WriterTimeoutWrapper struct {
 	t time.Duration
-	DataAvailabilityService
+	DataAvailabilityServiceWriter
+}
+
+type TimeoutWrapper struct {
+	ReaderTimeoutWrapper
+	WriterTimeoutWrapper
 }
 
-func NewTimeoutWrapper(dataAvailabilityService DataAvailabilityService, t time.Duration) DataAvailabilityService {
-	return &TimeoutWrapper{
-		t:                       t,
-		DataAvailabilityService: dataAvailabilityService,
+func NewReaderTimeoutWrapper(dataAvailabilityServiceReader DataAvailabilityServiceReader, t time.Duration) DataAvailabilityServiceReader {
+	return &ReaderTimeoutWrapper{
+		t:                             t,
+		DataAvailabilityServiceReader: dataAvailabilityServiceReader,
+	}
+}
+func NewWriterTimeoutWrapper(dataAvailabilityServiceWriter DataAvailabilityServiceWriter, t time.Duration) DataAvailabilityServiceWriter {
+	return &WriterTimeoutWrapper{
+		t:                             t,
+		DataAvailabilityServiceWriter: dataAvailabilityServiceWriter,
 	}
 }
 
-func (w *TimeoutWrapper) GetByHash(ctx context.Context, hash common.Hash) ([]byte, error) {
+func (w *ReaderTimeoutWrapper) GetByHash(ctx context.Context, hash common.Hash) ([]byte, error) {
 	deadlineCtx, cancel := context.WithDeadline(ctx, time.Now().Add(w.t))
 	// For Retrieve we want fast cancellation of all goroutines started by
```

Comment:
Should this comment say "GetByHash" instead of "Retrieve"?

Reply:
Updated comment.